slip-0039.md

```diff
@@ -69,17 +69,18 @@ This structure is then converted into a mnemonic passphrase by splitting it up b
 
 ## Checksum
 
-For checksum we use 16-bits of SHA-256 hash function.
+For the checksums we use the leftmost 16 bits of a SHA-256 hash digest of the relevant payload.
 
 ## Passphrase
 
 When enough M secrets are provided the master secret is reconstructed. To allow an additional protection of the final seed using a passphrase we suggest a following scenario. The passphrase is stretched using PBKDF2, which servers as a key to an AES encryption. The result of this encryption is the final seed to be used. If no passphrase is provided the passphrase is set to an empty string.
+(TODO: Use domain separation in PBKDF2)
```

Comment:
What exactly do you mean by this?

Reply:
@prusnak Presumably some sort of salt

Reply:
Yeah, so I am not really sure about this. My gut tells me it would be a good idea to include a salt here, to make it secure against rainbow-table attacks. I am not really sure though.

Reply:
@dsprenkels ACK regarding a salt. We're only using this as a symmetric encryption key, so it cannot hurt to have a salt since the user can still retrieve the master secret and switch to another standard, if they wish.

Reply:
Included fixed salt in the spec for now.

Reply:
I am also not 100% sure on this, but I believe we need salt in case the master secret is some kind of predictable value (like passwords or dictionary words), but it is not required in our case where the master secret is the wallet entropy, which is required to be random on its own.

One way or another, the PBKDF2 requires salt anyway and it definitely doesn't hurt to use it. In the traditional case of storing passwords in a database, you should use per-password unique salts to prevent mass computation of all the passwords. Since this is not our case I believe using a hard-coded string as a salt is fine. Correct me if I'm wrong.

Reply:
> I believe we need salt in case the master secret is some kind of predictable value

@tsusanka If that's the case, bear in mind that this standard could be used for most kinds of data (including those which may not be completely random).

Reply:
@saleemrashid you're right. Giving it another thought the salt being random would be the best scenario, but a user would have to remember some additional random data and I'm not sure we want to go this way. I have to think more about this :)

Reply:
@saleemrashid I think I'm wrong, it doesn't make sense. Is it okay if I delete our comments?

Reply:
@tsusanka It's up to you :smile: